GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -27,7 +27,7 @@
 
 namespace GitUI.CommandsDialogs
 {
-    public sealed partial class FormCommit : GitModuleForm
+    public sealed partial class FormCommit : GitModuleForm, IRunScript.IFileListSource
```

Comment:
`Edit File`, `Delete File` and even `Open in Visual Studio` are available in this file list. I see no reason why scripts should be the exception.

Also, while it's possible to close the dialog and use the artificial commits instead, that is quite inconvenient. It wouldn't be a problem if the dialog was not modal, and the artificial commits could be accessed without closing it.

Reply:
"Edit file" and "Delete file" are quite valid in this context - a changed file could have been incorrectly changed or auto-generated.
"Open in VS" shouldn't be available here.

I am unconvinced the scripts are necessary in this dialog. Can you please describe a use case that would absolute require the scripts here?

Reply:
Scripts should be here, for sure for worktree but I see no reason to not include index.
You often have to adjust files when staging which could require editing, also in vs code.

Reply:
If "Edit file" is valid, then so are scripts, because you can have an "Edit file" script that opens a custom editor (which is what started this whole thing). Can also have scripts for deleting files in a certain way (e.g. moving them to a quarantine folder instead of outright deleting).

Another use case: Running a lint formatter on a file. You could, during the commit, notice a file is not formatted properly. If you had set up a formatter as a user script, you can quickly run it, and see the result.